lib/api/bucketPutReplication.js

```diff
@@ -0,0 +1,74 @@
+import { waterfall } from 'async';
+import { errors } from 'arsenal';
+
+import metadata from '../metadata/wrapper';
+import { metadataValidateBucket } from '../metadata/metadataUtils';
+import { pushMetric } from '../utapi/utilities';
+import { getReplicationConfiguration } from
+    './apiUtils/bucket/bucketReplicationXMLModel';
+import collectCorsHeaders from '../utilities/collectCorsHeaders';
+
+// Check that versioning is 'Enabled' on the given bucket.
+function isVersioningEnabled(bucket) {
+    const versioningConfig = bucket.getVersioningConfiguration();
+    return versioningConfig ? versioningConfig.Status === 'Enabled' : false;
+}
+
+// The error response when a bucket does not have versioning 'Enabled'.
+const invalidVersioningError = errors.InvalidRequest.customizeDescription(
+    'Versioning must be \'Enabled\' on the bucket to apply a replication ' +
+    'configuration');
+
+/**
+ * bucketPutReplication - Create or update bucket replication configuration
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+export default function bucketPutReplication(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketPutReplication' });
+    const { bucketName, post, headers, method } = request;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: 'bucketPutReplication',
```

Comment:
No special authorization—just need to be the bucket owner to put the configuration.